bin/BcoinClient.js

```diff
@@ -0,0 +1,281 @@
+
+    if (tx.height < 0)
+      throw new Error('Transaction not confirmed');
+
+    const json = await this.getBlockHeader(tx.height);
+
+    if (!json)
+      throw new Error('Cannot find header');
+
+    const header = await this.getHeader(tx.height, enc);
+
+    const txinfo = parseTxHex(tx.hex);
+
+    let [nodes, index] = await this.getMerkleProof(txid, tx.height);
+
+    let path = '';
+    for (let node of nodes)
+      path += node;
+
+    return {
+      version: txinfo.version,
+      vin: txinfo.vin,
+      vout: txinfo.vout,
+      locktime: txinfo.locktime,
+      tx_id: txid,
+      tx_id_le: reverse(txid),
+      index: index,
+      confirming_header: header,
+      intermediate_nodes: path
+    }
+  }
+
+  /**
+   * Fetch a header by height or hash.
+   */
+
+  async getHeader(height) {
+    const json = await this.getBlockHeader(height);
+
+    if (!json)
+      return null;
+
+    const hex = await this.execute('getblockheader', [json.hash, false]);
+
+    return {
+      raw: hex,
+      hash: json.hash,
+      hash_le: reverse(json.hash),
+      height: height,
+      prevhash: json.prevBlock,
+      merkle_root: json.merkleRoot,
+      merkle_root_le: reverse(json.merkleRoot)
+    }
+  }
+
+  /**
+   * Validate the merkle tree of a block and then compute
+   * a merkle proof of inclusion for the txid.
+   * @param {String} txid
+   * @param {Number} height
+   * @returns {[][]String, Number} - a merkle proof and the index of the leaf
+   */
+
+  async getMerkleProof(txid, height) {
+    const block = await super.execute('getblockbyheight', [height]);
+
+    let index = -1;
+    const txs = [];
+    for (const [i, tx] of Object.entries(block.tx)) {
+      if (tx === txid)
+        index = i >>> 0; // cast to uint from string
+      txs.push(Buffer.from(tx, 'hex').reverse());
+    }
+
+    assert(index >= 0, 'Transaction not in block.');
+
+    const [root] = merkle.createRoot(hash256, txs.slice());
+    assert.bufferEqual(Buffer.from(block.merkleroot, 'hex').reverse(), root);
+
+    const branch = merkle.createBranch(hash256, index, txs.slice());
+
+    const proof = [];
+    for (const hash of branch)
+      proof.push(hash.toString('hex'));
+
+    return [proof, index];
+  }
+
+  /**
+   * Fetch a header chain by count.
+   * @param {Number} height - starting block height
+   * @param {Number} count - number of headers
+   * @param {String} enc - json or hex
+   */
+
+  async getHeaderChainByCount(height, count, enc) {
+    assert((height >>> 0) === height);
+    assert((count >>> 0) === count);
+
+    const headers = [];
+
+    for (let i = 0; i < count; i++) {
+      const json = await this.getBlockHeader(height);
+
+      if (!json)
+        throw new Error('Cannot find header');
+
+      if (enc === 'json') {
+        headers.push(json);
+      } else {
+        const hex = await this.execute('getblockheader', [json.hash, false]);
+        headers.push(hex);
+      }
+    }
+
+    if (enc === 'hex')
```

Comment:
This should probably return `{headers: headers.join('')}` so that return types are consistently json

Reply:
for line 168 and 170?

Reply:
yup

Reply:
Yeah. Same return type is 👍